### PR TITLE
Make collection data entry page fully editable

### PR DESCRIPTION
**Pull request by Gbarra9**
The pull request addresses the issue:

- https://github.com/episphere/connect/issues/595

The following changes were made:

**Issue related**
- enabled editButton to enable Sample Collected checkbox and Reason Not Collection input field

**Code Issue Related**

*Enable field for issue

- added [field enabling logic](https://github.com/episphere/biospecimen/pull/553/files#diff-aa3209e98840e6d236562086e3c6942a28f0c6beef6897d036367d09cc313c59R2463-R2467) for edit button

*Toggle related 
-  added `reasonDropdown` to `addEventBiospecimenCollectionFormToggles` function to prevent user from updating biospecimen object with a collected id checkbox yes value and a reasonNotCollected value 
- added `reasonDropdown` to `addEventBiospecimenCollectionFormToggles` function to clear checkbox and disable/clear fields not available when someone selects a value from the reason not collected dropdown
- added condition `if (deviation.checked && comment.disabled) comment.disabled = false;` in the `addEventBiospecimenCollectionFormToggles` function for deviated checkboxes to enable comment field

**Extra**
- renamed data parameters and arguments from `formData` to `biospecimenData`
- replaced concept id numbers with either existing or new concept id keys for readability
- renamed key names fName, lName, mName in `fieldToConceptIdMapping.js`
- added key `deviationType` to `collection` key object in `fieldToConceptIdMapping.js`
- changed updateParticipant parameter `array` to `dataObj` because function call is passing an object

**Side Notes (optional read)**
- On the collection Data Entry Page the `edit button` appears if the tube/bag is not collected and not finalized 
- The `finalized - 410912345` is saved to the biospecimen object once the user goes to the Review Collection Data Entry
page and confirms the collection
- No edit buttons will appear if the biospecimen object has the finalized key and yes value. This is true in the scenario where someone saved a tube not collected yet on the collection data entry page but the other items have finalized in the past. A user cannot edit the saved tube in the Collection Data Entry page in this scenario.
- Checking the collected checkbox enables the Scan Full Specimen ID input field and the Select For Deviation check box, and disables and clears the value for Reason Not Collected dropdown

**Images**

On load 
Screenshot 2023-07-14 at 3 48 04 AM

User clicks on edit button for Serum Separator Tube (0001)
Screenshot 2023-07-14 at 3 48 19 AM

User selects a Reason Not Collected
Screenshot 2023-07-14 at 3 49 05 AM

### CODE REVIEW

File: src/events.js [hunk 1]
```diff
@@ -2408,14 +2407,48 @@ export const addEventBiospecimenCollectionFormToggles = () => {
...
+        });
+    });
+
+    reasonNotCollectedDropdown.forEach( reasonDropdown => {
+        const collectedId = document.getElementById(reasonDropdown.id).id.replace('Reason', '');
+        const collected = document.getElementById(collectedId);
+        const specimenId = document.getElementById(`${collectedId}Id`);
+        const deviation = document.getElementById(`${collectedId}Deviated`);
+        const type = document.getElementById(`${collectedId}Deviation`);
+        const comment = document.getElementById(`${collectedId}DeviatedExplanation`);
```
Comment (JoeArmani) [inline src/events.js:2428]: Nice!

File: src/events.js [hunk 2]
```diff
@@ -2550,37 +2591,34 @@ const collectionSubmission = async (formData, biospecimenData, cntd) => {
...
+        } else if (input.required && (tubes[0].id !== tubeID && !additionalTubeIDRequirement.regExp.test(tubeID))) {
             hasError = true;
             hasCntdError = true;
             errorMessage(input.id, 'Invalid Full Specimen ID.', focus);
             focus = false;
         }
 
-        if (input.required) biospecimenData[`${input.id.replace('Id', '')}`]['825582494'] = `${masterID} ${tubeID}`.trim();
+        if (input.required) biospecimenData[`${input.id.replace('Id', '')}`][conceptIds.collection.tube.scannedId] = `${masterID} ${tubeID}`.trim();// here
     });
```
Comment (JoeArmani) [inline src/events.js:2622]: is this comment a remnant from dev?

**Review by JoeArmani — APPROVED** [review]
Huge readability improvement. Nice work!

-confirmed all added conceptIds match data dictionary.
-potentially one comment left over from dev? No actual concerns.

looks good

File: src/events.js (hunk 2 above)
Comment (Gbarra9) [reply src/events.js:2622]: Thanks, i'll remove it

File: src/events.js [hunk 3]
```diff
@@ -2426,19 +2459,27 @@ export const addEventBiospecimenCollectionFormEdit = () => {
         button.addEventListener('click', () => {
             const conceptID = button.id.replace('collectEditBtn', '');
             document.getElementById(conceptID + 'Id').disabled = false;
+            const tubeCollectedCheckbox = document.getElementById(conceptID);
+            if (tubeCollectedCheckbox) tubeCollectedCheckbox.disabled = false;
+            const reasonNotCollectedDropdown = document.getElementById(conceptID + 'Reason');
+            if (reasonNotCollectedDropdown) reasonNotCollectedDropdown.disabled = false;
+
```
Comment (we-ai) [inline src/events.js:2466]:
```suggestion
```

File: src/events.js [hunk 4]
```diff
@@ -2426,19 +2459,27 @@ export const addEventBiospecimenCollectionFormEdit = () => {
...
 
-                if(deviation.checked) {
+                if (deviation.checked) {
                     const type = document.getElementById(deviation.id.replace('Deviated', 'Deviation'));
                     const comment = document.getElementById(deviation.id + 'Explanation'); 
 
                     type.disabled = false;
                     comment.disabled = false;
                 }
+
```
Comment (we-ai) [inline src/events.js:2479]:
```suggestion
```

**Review by we-ai — APPROVED** [review]
Looks good. 
Thanks for replacing hard-coded concept IDs with more readable names.

**Pull request merged**